http-filter-example/http_filter.cc

```diff
@@ -1,18 +1,60 @@
 #include <string>
+#include <unordered_map>
+#include <sstream>
+#include <vector>
 
 #include "http_filter.h"
 
+#include "source/common/common/logger.h"
 #include "envoy/server/filter_config.h"
 
 namespace Envoy {
 namespace Http {
 
 HttpSampleDecoderFilterConfig::HttpSampleDecoderFilterConfig(
     const sample::Decoder& proto_config)
-    : key_(proto_config.key()), val_(proto_config.val()) {}
+    : key_(proto_config.key()), val_(proto_config.val()), extra_(stoi(proto_config.extra())) {}
 
 HttpSampleDecoderFilter::HttpSampleDecoderFilter(HttpSampleDecoderFilterConfigSharedPtr config)
-    : config_(config) {}
+    : config_(config) {
+  // populate list of valid operations, implementation will be changed in the next few commits
+  operations_.insert("set-header");
+  
+  const std::string header_config = headerValue();
+
+  // find the position of the first space character
+  const size_t spacePos = header_config.find(' ');
+
+  const auto operation = header_config.substr(0, spacePos);
+
+  // error checking -- invalid operation
+  auto it = operations_.find(operation);
```

Comment:
Nitpick: can be const, or even `if (operations_.find(operation) == operations_.end()) {`